### PR TITLE
Allow call of `endMeasure` from anywhere

### PR DESCRIPTION
**Pull request by mxdvl**
## What are you changing?

- add a method to `endMeasure` from anywhere

## Why?

- @arelra [suggested it could be worthwhile](https://github.com/guardian/csnx/pull/729#discussion_r1272118294)

### CODE REVIEW

File: libs/@guardian/libs/src/measure/startMeasure.ts [hunk 1]
```diff
@@ -31,16 +34,28 @@ export const startMeasure = (
...
-	const endMeasure = () => {
-		const measureName = serialise(team, name, action);
+	const measureName = serialise(team, name, action);
 
-		const { duration } = window.performance.measure(measureName, options) ??
+	const endMeasure = (end?: number) => {
+		const { duration } = window.performance.measure(measureName, {
+			...options,
+			end,
+		}) ??
```
Comment (arelra) [inline libs/@guardian/libs/src/measure/startMeasure.ts:43]: Just wondering when the LHS would this return nullish? According to the docs `measure` returns either a `PerformanceMeasure` or throws an exception.

File: libs/@guardian/libs/src/measure/startMeasure.ts [hunk 2]
```diff
@@ -31,16 +34,28 @@ export const startMeasure = (
...
 			window.performance.getEntriesByName(measureName, 'measure').at(-1) ?? {
 				duration: performance.now() - options.start,
 			};
 
 		return Math.ceil(duration);
 	};
 
-	return { endMeasure };
+	const observe = () =>
```
Comment (arelra) [inline libs/@guardian/libs/src/measure/startMeasure.ts:51]: I feel like the name could describe what the function does better. 

Just an idea but  `observeEndMark()`?

File: libs/@guardian/libs/src/measure/startMeasure.ts [hunk 3]
```diff
@@ -31,16 +34,28 @@ export const startMeasure = (
...
+		new PerformanceObserver((entries, observer) => {
+			const [endMark] = entries.getEntriesByName(measureName);
```
Comment (arelra) [inline libs/@guardian/libs/src/measure/startMeasure.ts:53]: This is cool 😎 . Just a suggestion re semantics and dev tools. Using the zeroth mark as the semantic end mark is a little unusual. I'm wondering if it is less surprising to anyone looking at the performance dev tools to always create start marks. Then we can grab the [1]'st mark as the end mark for `observe`.

File: libs/@guardian/libs/src/measure/startMeasure.ts (hunk 1 above)
Comment (mxdvl) [reply libs/@guardian/libs/src/measure/startMeasure.ts:43]: [Firefox before v101 returns `undefined`](https://github.com/guardian/csnx/pull/729/files#diff-d6ff5d412d9e790b174922bb617f2c51fc0df428a74cf235ab0df7c0b1dc6f49R4-R5)

**Comment by mxdvl** [comment]
This seems like a footgun, so let’s wait until there’s an actual need for such a method!

**Pull request closed without merging**